core/app/components/pubs/PubEditor/PubEditorClient.tsx

```diff
@@ -327,8 +327,9 @@ export const PubEditorClient = ({
 				buttonElements,
 			});
 
-			const newStageId = stageIdFromForm ?? stageIdFromButtonConfig;
+			const newStageId = stageIdFromButtonConfig ?? stageIdFromForm ?? undefined;
```

Comment:
It's a slightly awkward experience for admins that actually update the stage select since clicking the submit button will ignore what they've selected there now. But I see that you'd probably have to remove the stageId from the default values to make the other order work, so I think this is fine for now!